v0/sources/appsflyer/transform.js

```diff
@@ -32,6 +32,10 @@ function processEvent(event) {
       }
     });
 
+    if (message.context.device == undefined) {
```

Comment:
@esadek can we do the check like `if( !isObject(message.context.device) ) { message.context.device = {} }` just in case device is wrongly set to a non-object value in reuqest/user-transformation

Reply:
Also let's add a [input](https://github.com/rudderlabs/rudder-transformer/blob/876b97353e045d59222cba7e13231057fa39951b/__tests__/data/af_input.json) and [output](https://github.com/rudderlabs/rudder-transformer/blob/876b97353e045d59222cba7e13231057fa39951b/__tests__/data/af_output.json) in the tests to verify

Reply:
I have changed the condition and added test data.